flink-streaming-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/AvroKeyValueSinkWriter.java

```diff
@@ -0,0 +1,308 @@
+		properties.put(AvroKeyValueSinkWriter.CONF_OUTPUT_KEY_SCHEMA, keySchema);
+		properties.put(AvroKeyValueSinkWriter.CONF_OUTPUT_VALUE_SCHEMA, valueSchema);
+		properties.put(AvroKeyValueSinkWriter.CONF_COMPRESS, Boolean.toString(true));
+		properties.put(AvroKeyValueSinkWriter.CONF_COMPRESS_CODEC, DataFileConstants.SNAPPY_CODEC);
+		
+		sink.setWriter(new AvroSinkWriter<Long , Long>(properties));
+		sink.setBatchSize(1024 * 1024 * 64); // this is 64 MB,
+}
+</pre>
+*/
+public class AvroKeyValueSinkWriter<K, V> extends StreamWriterBase<Tuple2<K, V>>  implements Writer<Tuple2<K, V>>, InputTypeConfigurable {
+	private static final long serialVersionUID = 1L;
+	public static final String CONF_OUTPUT_KEY_SCHEMA = "avro.schema.output.key";
+	public static final String CONF_OUTPUT_VALUE_SCHEMA = "avro.schema.output.value";
+	public static final String CONF_COMPRESS = FileOutputFormat.COMPRESS;
+	public static final String CONF_COMPRESS_CODEC = FileOutputFormat.COMPRESS_CODEC;
+	public static final String CONF_DEFLATE_LEVEL = "avro.deflate.level";
+	public static final String CONF_XZ_LEVEL = "avro.xz.level";
+
+	private transient AvroKeyValueWriter<K, V> keyValueWriter;
+
+	private final Map<String, String> properties;
+
+	/**
+	 * C'tor for the writer
+	 * <p>
+	 * You can provide different properties that will be used to configure avro key-value writer as simple properties map(see example above)
+	 * @param properties
+	 */
+	public AvroKeyValueSinkWriter(Map<String, String> properties) {
+		this.properties = properties;
+	}
+
+	private boolean getBoolean(Map<String,String> conf, String key, boolean def) {
+		String value = conf.get(key);
+		if (value == null) {
+			return def;
+		}
+		return Boolean.parseBoolean(value);
+	}
+	
+	private int getInt(Map<String,String> conf, String key, int def) {
+		String value = conf.get(key);
+		if (value == null) {
+			return def;
+		}
+		return Integer.parseInt(value);
+	}
+
+	//this derived from AvroOutputFormatBase.getCompressionCodec(..)
+	private CodecFactory getCompressionCodec(Map<String,String> conf) {
+		if (getBoolean(conf, CONF_COMPRESS, false)) {
+			int deflateLevel = getInt(conf, CONF_DEFLATE_LEVEL, CodecFactory.DEFAULT_DEFLATE_LEVEL);
+			int xzLevel = getInt(conf, CONF_XZ_LEVEL, CodecFactory.DEFAULT_XZ_LEVEL);
+
+			String outputCodec = conf.get(CONF_COMPRESS_CODEC);
+
+			if (DataFileConstants.DEFLATE_CODEC.equals(outputCodec)) {
+				return CodecFactory.deflateCodec(deflateLevel);
+			} else if (DataFileConstants.XZ_CODEC.equals(outputCodec)) {
+				return CodecFactory.xzCodec(xzLevel);
+			} else {
+				return CodecFactory.fromString(outputCodec);
+			}
+		}
+		return CodecFactory.nullCodec();
+	}
+
+	@Override
+	public void open(FileSystem fs, Path path) throws IOException {
+		super.open(fs, path);
+
+		CodecFactory compressionCodec = getCompressionCodec(properties);
+		  		  
+		String keySchemaString = properties.get(CONF_OUTPUT_KEY_SCHEMA);
+		if (keySchemaString == null) {
```

Comment:
I think it makes more sense to do that check in the constructor. Otherwise, the code will fail after the deployment.